internal/restore/volumesnapshot_action.go

```diff
@@ -90,24 +87,18 @@ func (p *VolumeSnapshotRestoreItemAction) Execute(input *velero.RestoreItemActio
 		// Overwrite snaphandle if csi data-mover case is true
 		if util.DataMoverCase() {
 
-			vsrList := datamoverv1alpha1.VolumeSnapshotRestoreList{}
-			snapMoverClient, err := util.GetVolumeSnapshotMoverClient()
+			vsrList, err := util.GetVolumeSnapshotRestoreWithStatusData(input.Restore.Name, *vs.Spec.Source.PersistentVolumeClaimName, p.Log)
 			if err != nil {
-				return nil, err
+				return nil, errors.WithStack(err)
 			}
 
-			VSRListOptions := client.MatchingLabels(map[string]string{
-				velerov1api.RestoreNameLabel:    input.Restore.Name,
-				util.PersistentVolumeClaimLabel: *vs.Spec.Source.PersistentVolumeClaimName,
-			})
-
-			err = snapMoverClient.List(context.TODO(), &vsrList, VSRListOptions)
-			if err != nil {
-				return nil, errors.Wrapf(err, fmt.Sprintf("failed to get volumesnapshotrestore for PVC %s", *vs.Spec.Source.PersistentVolumeClaimName))
+			// this will always be unique
+			if len(vsrList.Items) > 0 {
```

Comment:
Do we need this check here since it is checked in the func called above?

Reply:
yeah so this will be more like the polling on the vsr timed out and we really got an empty list